### PR TITLE
Jakarta Expression Language 3.0

### PR DESCRIPTION
**Pull request by arjantijms**
Signed-off-by: arjantijms <arjan.tijms@gmail.com>

## Specification PR template
When creating a specification project release review, create two PRs with the content divided as follows.

Include the following in PR#1:
- [X] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [X] Specification PDF in the form of wombat_x.y.pdf
- [X] Specification HTML in the form of wombat_x.y.html
- [X] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] For a Release Review, a summary that a Compatible Implementation
      is complete, passes the TCK, and that the TCK includes sufficient
      coverage of the specification. The TCK users guide MUST include
      the instructions to run the compatible implementations used to
      validate the release.  Instructions MAY be by reference.
- [X] The URL of the OSSRH staging repository for the api, javadoc:
      https://oss.sonatype.org/content/repositories/staging/jakarta/el/jakarta.el-api/3.0.3/
- [X] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/eclipse-expression-language-tck-3.0.0.zip
- [X] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/el-ri/issues/103

### CODE REVIEW

**Comment by dblevins** [comment]
Spec status should be "Final" or "Final Release"

File: expression-language/3.0/_index.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+|Red Hat                |         |         |          |
+|Tomitribe              |         |         |          |
+|EE4J PMC               |         |         |          |
+|Participant Members    |         |         |          |
+|Committer Members      |         |         |          |
+
+
+# Compatible Implementations
+
+* Expression Language RI 3.0.3
```
Comment (bshannon) [inline expression-language/3.0/_index.md]: There are no longer "Reference Implementations", so this Compatible Implementation
needs to use something other than "RI".  If you don't have a clever project name,
consider something like "EE4J Implementation of Expression Language".

Also, this should include a link to the implementation project.

File: expression-language/3.0/tck_results_summary.txt [hunk 2]
```diff
@@ -0,0 +1,344 @@
+[javatest.batch] Number of Tests Passed      = 336
```
Comment (bshannon) [inline expression-language/3.0/tck_results_summary.txt]: The results summary needs to be published on a page related to the implementation project,
not here.

File: expression-language/3.0/_index.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+---
+title: "Jakarta Expression Language Specification 3.0"
+date: 2019-07-22T16:45:02+0200
```
Comment (bshannon) [inline expression-language/3.0/_index.md]: The time component is not required, and probably not useful in this context.

File: expression-language/3.0/_index.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+title: "Jakarta Expression Language Specification 3.0"
+date: 2019-07-22
+summary: "First release for Jakarta EE 8"
+---
+Jakarta Expression Language defines an expression language for Java applications
```
Comment (bshannon) [inline expression-language/3.0/_index.md]: Need period at end of sentence.

File: expression-language/3.0/_index.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+* [Jakarta Expression Language 3.0 Specification Document](./expression-language_3.0.pdf) (PDF)
+* [Jakarta Expression Language 3.0 Specification Document](./expression-language_3.0.html) (HTML)
+* [Jakarta Expression Language 3.0 Javadoc](./apidocs)
```
Comment (bshannon) [inline expression-language/3.0/_index.md]: Need link to the final TCK.

File: expression-language/3.0/_index.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+|Red Hat                |         |         |          |
+|Tomitribe              |         |         |          |
+|EE4J PMC               |         |         |          |
+|Participant Members    |         |         |          |
+|Committer Members      |         |         |          |
+
+
+# Compatible Implementations
+
+* Expression Language RI 3.0.3
```
Comment (bshannon) [inline expression-language/3.0/_index.md]: I don't think we should call it an "RI", and we need a link to the project.

**Comment by arjantijms** [comment]
@bshannon 

I'm not 100% sure what this is referring to:

>in the spec docs, the Specification: header should not have the version number and the Version: header should be only "3.0"

Do you mean that:

```
 Specification: Jakarta Expression Language 3.0
  Version: 3.0.3
  Status: Final
  Release: July 25, 2019
```

Should become:

```
 Specification: 3.0
  Version: 3.0.3
  Status: Final
  Release: July 25, 2019
```

**Comment by bshannon** [comment]
No, it should become:

```
Specification: Jakarta Expression Language
Version: 3.0
Status: Final
Release: July 25, 2019
```

**Comment by arjantijms** [comment]
> No, it should become:

I see, thanks Bill, I'll sort it.

**Comment by starksm64** [comment]
For wave 1, this spec should be due by 8/7, so please review this ASAP. Note there is a good checklist for https://github.com/jakartaee/specification-committee/blob/master/spec_review_checklist.md

File: expression-language/3.0/_index.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+title: "Jakarta Expression Language Specification 3.0"
+date: 2019-08-07
+summary: "First release for Jakarta EE 8"
+---
+Jakarta Expression Language defines an expression language for Java applications.
+
+* [Jakarta Expression Language 3.0 Specification Document](./expression-language-spec-3.0.pdf) (PDF)
+* [Jakarta Expression Language 3.0 Specification Document](./expression-language-spec-3.0.html) (HTML)
+* [Jakarta Expression Language 3.0 Javadoc](./apidocs)
+* [Jakarta Expression Language 3.0 TCK](http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/eclipse-expression-language-tck-3.0.0.zip)
```
Comment (starksm64) [inline expression-language/3.0/_index.md]: Format of link needs to be https://download.eclipse.org/jakartaee/expression-language/x.y/expression-language-tck-x.y.z.zip

**Comment by starksm64** [comment]
# Spec Review Checklist

1. Spec PR
  - [x] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [x] Directory of form {spec}/x.y
  - [x] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [x] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [x] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [x] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [x] Staging repository link of the form https://oss.sonatype.org/conte/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [x] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [x] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [x] Second PR for just apidocs

2. _index.md
  - [x] Link to spec pdf
  - [x] Link to spec html
  - [x] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [x] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [x] Name of and link to at least one Compatible Implementation

3. javadocs 
  - [] Footer contains Eclipse copyright and link to license (apidocs has this, staged javadocs don't)
  - [x] ESFL license is included, usually as doc-files/speclicense.html
  - [x] no META-INF directory in PR

4. Spec PDF
  - [x] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z (has 3.0.3)
  - [x] Correct Eclipse copyright line
  - [x] No DRAFT or SNAPSHOT

5. Spec HTML
  - [x] Same as PDF

6. TCK zip file
  - [x] README file (optional for this release) (in docs/README.txt)
  - [x] EFTL license file, preferably named LICENSE.md
  - [x] User's Guide (or equivalent documentation)
  - [x] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [x] Software requirements listed
  - [x] Installation and configuration described
  - [x] How to run tests
  - [x] Where to file challenges

8. Compatibility certification request
  - [x] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [x] SHA-256 fingerprint matches staged TCK zip file
7af99b28c81c1f7eba1e241b20f3854289ae511c8446158671f32731c785504c
  - [x] Request issue has `certification` label.

9. TCK results summary
  - [x] Page is hosted by Compatible Implementation project
  - [x] Includes all information from certification request
  - [x] Summary includes number of tests passed, failed, errors

**Comment by arjantijms** [comment]
> Although the apidocs html files has a link to the license, when I looked at the https://oss.sonatype.org/content/repositories/staging/jakarta/el/jakarta.el-api/3.0.3/jakarta.el-api-3.0.3-javadoc.jar contents, those files did not. This is optional for this release.

It has now:

javadoc-license

**Comment by starksm64** [comment]
> It has now:
> 
> javadoc-license

I do see the speclicense.html in the javadoc-jar, but the footer is not linking to it. Again, not required for this release.

**Comment by kwsutter** [comment]
* On ballot completion
 - [x] The designated specification committee representative adds this final checklist to the main PR.
 - [ ] The specification committee representative merges the specification PRs.
 - [ ] The designated specification committee representative calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job.
 - [ ] The specification project member who created the staging release promotes the specification api jars to maven central.
 - [ ] The EMO updates the specification page with the ballot results.
This list goes on the committed spec index page.
 - [ ] The specification project team should go through the merged spec website page to verify all the links are valid.
 - [ ] The specification project team should approve the compatibility request and send an email to tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] The specification project team should merge any final release branch as approriate for the branch management for the project.

**Comment by arjantijms** [comment]
@kwsutter 
Staging release (as well as staged implementation) promoted to maven central:

* https://repo1.maven.org/maven2/jakarta/el/jakarta.el-api/3.0.3/
* https://repo1.maven.org/maven2/org/glassfish/jakarta.el/3.0.3/

**Comment by bshannon** [comment]
Thanks @arjantijms, can you verify that the last two items in the [checklist](https://github.com/jakartaee/specifications/pull/40#issuecomment-526606193) were done?

**Comment by arjantijms** [comment]
@bshannon 

* Mail sent
* Branch merged: https://github.com/eclipse-ee4j/el-ri/pull/105

**Pull request merged**